client/packages/invoices/src/Returns/InboundDetailView/Toolbar.tsx

```diff
@@ -49,17 +56,14 @@ export const Toolbar: FC = () => {
               label={t('label.customer-ref')}
               Input={
                 <BasicTextInput
-                  disabled={true}
                   size="small"
                   sx={{ width: 250 }}
-                  value={data?.theirReference ?? ''}
-                  // TODO: once updating theirReference supported by API
-                  // disabled={isDisabled}
-                  // value={theirReferenceBuffer ?? ''}
-                  // onChange={event => {
-                  //   setTheirReferenceBuffer(event.target.value);
-                  //   update({ theirReference: event.target.value });
-                  // }}
+                  disabled={isDisabled}
+                  value={theirReferenceBuffer ?? ''}
+                  onChange={event => {
+                    setTheirReferenceBuffer(event.target.value);
```

Comment:
I've noticed, for the comment in AdditionalInfoSection we don't use state variable for value, and together with debouncing on a slow network (slow 3g throttle), I get this:

https://github.com/msupply-foundation/open-msupply/assets/26194949/40a010b7-0b9c-4da6-8a31-fd48395f2c40

Do you know if this is the same behaviour for other side panels (or any other inputs you've encountered, where we use value = {something from graphql} and then update = {update graphql} ?)

Reply:
Oh ew haha, thanks for spotting. I've not used the `fields` hook for returns, but it appears I missed making it NOT do this. Will have a quick play... I like the debouncedState all in one idea 🤔

Reply:
Thoughts on this version? https://github.com/msupply-foundation/open-msupply/compare/3316-allow-mutation-of-more-invoice-fields...3316-allow-mutation-of-more-invoice-fields(useEffect-buffer-state-version)

Reply:
Thanks, I think this is good for now, until we think of a pattern to do this. Just my quick 2 cents, I reckon the API abstractions should focus on easy 'consumption' rather then logical structure/partitioning. Like I would love a hook like this (just. a quick brainstorm)

```typescript
// result is a union (rather then checking if it's loading and then if it's error or data etc..
const { patch, result } = useOutboundShip("thisIsTheId");
// do the error/loading checking etc..
const { theirReference, comment } = result.shipment
// .... that field
value={theirReference}
onChange={e => {
   // patch sets buffered state and queues debounce etc..
   patch({theirReference: e.target.value})
}}
```

Reply:
I like this, with the data and mutation coming from the same hook - clean and easy to reason about. Would be keen to come back to it